src/inc.c

```diff
@@ -555,12 +555,19 @@ void do_incremental_crack(struct db_main *db, char *mode)
 #endif
 	}
 
+	if (options.req_minlength >= 0 && min_length > max_length &&
+	   (options.req_maxlength == 0 || maxlength_computed) &&
+	    options.req_minlength <= CHARSET_LENGTH) {
+		max_length = min_length;
```

Comment:
I wouldn't silently adjust max_length.
At least the user must be informed about such an adjustment (at default verbosity)

Just aborting with an error message telling the user what to do might be better.

Reply:
It is fair, but, in my opinion, if I say `--min-len=X`, at least, I'm saying `--max-len=X` (it could be greater). 
* So, I dislike the idea to quit with error.

Reply:
OTOH, I dislike wasting CPU time on something I didn't really intend to do.

So it all depends. If the min length vs. max length check occurs after john loaded millions of hashes, then automatically adjusting the max length to the smallest possible value might be preferable.

But when the max length is checked first, I would still prefer an error message.

Let's wait for opinions of other people.